tarofreighter/parcel.go

```diff
@@ -571,55 +301,54 @@ func (s *sendPackage) createProofs() (spendProofs, error) {
 		}
 
 		err := proof.AddExclusionProofs(
-			&senderParams.BaseProofParams, s.FundedPkt.Pkt,
-			isAnchor,
+			&senderParams.BaseProofParams,
+			s.AnchorTx.FundedPsbt.Pkt, isAnchor,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("error adding exclusion proof "+
-				"for change output: %w", err)
+			return nil, nil, fmt.Errorf("error adding exclusion "+
+				"proof for change output: %w", err)
 		}
 
 		err = proof.AddExclusionProofs(
-			&receiverParams.BaseProofParams, s.FundedPkt.Pkt,
-			isAnchor,
+			&receiverParams.BaseProofParams,
+			s.AnchorTx.FundedPsbt.Pkt, isAnchor,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("error adding exclusion proof "+
-				"for change output: %w", err)
+			return nil, nil, fmt.Errorf("error adding exclusion "+
+				"proof for change output: %w", err)
 		}
 	}
 
 	senderProof, err := proof.CreateTransitionProof(
-		s.InputAsset.AnchorPoint, &senderParams,
+		inputPrevID.OutPoint, &senderParams,
 	)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	receiverProof, err := proof.CreateTransitionProof(
-		s.InputAsset.AnchorPoint, &receiverParams,
+		inputPrevID.OutPoint, &receiverParams,
 	)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return spendProofs{
-		asset.ToSerialized(s.SenderScriptKey.PubKey):  *senderProof,
-		asset.ToSerialized(&s.ReceiverAddr.ScriptKey): *receiverProof,
-	}, nil
+	return senderProof, receiverProof, nil
 }
 
 // deliverResponse delivers a response for the parcel back to the receiver over
 // the specified response channel.
 func (s *sendPackage) deliverResponse(respChan chan<- *PendingParcel) {
-	oldRoot := s.InputAsset.Commitment.TapscriptRoot(nil)
+	oldRoot := s.InputCommitment.TapscriptRoot(nil)
 
+	senderAsset := s.VirtualPacket.Outputs[0].Asset
+	receiverAsset := s.VirtualPacket.Outputs[1].Asset
```

Comment:
Should check if our send was interactive here, otherwise this panics? Unless deferring that for follow-up PR.

Reply:
Yes, that'll be done in the follow-up PR.